### DISCUSSION TITLE
Easy way to add displayname for react components.

### DISCUSSION THREAD

**Discussion by clijiac** [web-infra-dev/rspack#4063] [Q&A] [answered]
- Tried the [https://github.com/runreflect/webpack-react-component-name](https://github.com/runreflect/webpack-react-component-name) but it seems use the **parser-hook** that rspack do not support now. 
- Also there is a swc plugin for handle this [https://github.com/hjkcai/swc-plugin-add-display-name](https://github.com/hjkcai/swc-plugin-add-display-name) . also **swc plugin** is not supported now.

For now I just use terser to minify to keep class name , but terser minify is quite slow
any easier workaround to support this?
```js
new minifyPlugin({
      minifier: "terser",
      keep_classnames: true, // any classes are probably React Components anyway
      //keep_fnames: /[A-Z].+/,
  }),
```

**Comment by hyf0** [comment, marked as answer]
swc minfier also has the `keepFnNames` option. https://swc.rs/docs/configuration/minification

Rspack use swc as the builtin minfier but it seems rspack doesn't expose this option.

https://github.com/web-infra-dev/rspack/blob/c5f1697a3eaecb11968dded35e159a3f39c81ca3/crates/node_binding/binding.d.ts#L708

**Reply by clijiac** [reply]
Also tried this one, but seems using built in swc-loader  with use swc minify 
will generate a quite large size file main(init) js, in my case 12mb vs 5mb.
but this can works. thank you

**Reply by clijiac** [reply]
never mind , the size different is caused by devFriendlySplitChunks true. after change it is good now
